src/include/access/slru.h

```diff
@@ -111,49 +112,49 @@ typedef struct SlruCtlData
 {
 	SlruShared	shared;
 
-	/*
-	 * Which sync handler function to use when handing sync requests over to
-	 * the checkpointer.  SYNC_HANDLER_NONE to disable fsync (eg pg_notify).
-	 */
-	SyncRequestHandler sync_handler;
-
-	/*
-	 * Decide whether a page is "older" for truncation and as a hint for
-	 * evicting pages in LRU order.  Return true if every entry of the first
-	 * argument is older than every entry of the second argument.  Note that
-	 * !PagePrecedes(a,b) && !PagePrecedes(b,a) need not imply a==b; it also
-	 * arises when some entries are older and some are not.  For SLRUs using
-	 * SimpleLruTruncate(), this must use modular arithmetic.  (For others,
-	 * the behavior of this callback has no functional implications.)  Use
-	 * SlruPagePrecedesUnitTests() in SLRUs meeting its criteria.
-	 */
+  /*
+   * Which sync handler function to use when handing sync requests over to
+   * the checkpointer.  SYNC_HANDLER_NONE to disable fsync (eg pg_notify).
+   */
+  SyncRequestHandler sync_handler;
+
+  /*
+   * Decide whether a page is "older" for truncation and as a hint for
+   * evicting pages in LRU order.  Return true if every entry of the first
+   * argument is older than every entry of the second argument.  Note that
+   * !PagePrecedes(a,b) && !PagePrecedes(b,a) need not imply a==b; it also
+   * arises when some entries are older and some are not.  For SLRUs using
+   * SimpleLruTruncate(), this must use modular arithmetic.  (For others,
+   * the behavior of this callback has no functional implications.)  Use
+   * SlruPagePrecedesUnitTests() in SLRUs meeting its criteria.
+   */
 	bool		(*PagePrecedes) (int, int);
 
-	/*
-	 * Dir is set during SimpleLruInit and does not change thereafter. Since
-	 * it's always the same, it doesn't need to be in shared memory.
-	 */
+  /*
+   * Dir is set during SimpleLruInit and does not change thereafter. Since
+   * it's always the same, it doesn't need to be in shared memory.
+   */
 	char		Dir[64];
 } SlruCtlData;
 
 typedef SlruCtlData *SlruCtl;
 
 typedef bool (*slru_kind_check_hook_type) (SlruCtl ctl);
 typedef bool (*slru_page_exists_hook_type) (SlruCtl ctl, int segno, off_t offset);
-typedef bool (*slru_read_page_hook_type) (SlruCtl ctl, int segno, off_t offset, char *buffer);
+typedef bool (*slru_read_page_hook_type) (SlruCtl ctl, int segno, off_t offset, XLogRecPtr min_lsn, char *buffer);
```

Comment:
Nevermind this comment. I saw your other PR.

Reply:
The other PR handles it for slru_read_page_hook_type. 
I didn't make any changes to slru_page_exists_hook_type because we only do that in case we need to create a segment file and that will only be called for the home region. Remote regions are always read_only.